guidance/guidance_los/scripts/los_guidance_action_server.py

```diff
@@ -0,0 +1,367 @@
+#!/usr/bin/env python3
+
+import os
+from dataclasses import dataclass
+
+import numpy as np
+import rclpy
+import yaml
+from geometry_msgs.msg import PoseStamped, Vector3Stamped
+from guidance_los.los_guidance_algorithm import ThirdOrderLOSGuidance
+from nav_msgs.msg import Odometry
+from rclpy.action import ActionServer
+from rclpy.action.server import ServerGoalHandle
+from rclpy.callback_groups import ReentrantCallbackGroup
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from std_msgs.msg import String
+from transforms3d.euler import quat2euler
+from vortex_msgs.action import NavigateWaypoints
+from vortex_msgs.msg import LOSGuidance
+
+
+@dataclass(slots=True)
+class State:
+    """Data class representing the current state of the vehicle."""
+
+    x: float = 0.0
+    y: float = 0.0
+    z: float = 0.0
+    yaw: float = 0.0
+    pitch: float = 0.0
+
+    def as_ndarray(self) -> np.ndarray:
+        """Convert state to numpy array for guidance calculations."""
+        return np.array([self.x, self.y, self.z, self.yaw, self.pitch])
+
+
+class GuidanceActionServer(Node):
+    """ROS2 Action Server node for waypoint navigation using LOS guidance."""
+
+    def __init__(self):
+        super().__init__('guidance_action_server')
+
+        # Get the configuration file path from parameters
+        config_path = self.declare_parameter('config_path', '').value
+
+        # Load configuration
+        if config_path and os.path.exists(config_path):
+            self.get_logger().info(f"Loading parameters from: {config_path}")
+            try:
+                with open(config_path) as file:
+                    config_params = yaml.safe_load(file)
+                    node_params = config_params.get('los_guidance_node', {}).get(
+                        'ros__parameters', {}
+                    )
+
+                    # Extract different parameter groups
+                    self.los_params = node_params.get('los_guidance', {})
+                    self.topic_params = node_params.get('topics', {})
+                    self.qos_params = node_params.get('qos', {})
+
+                    self.get_logger().info("Configuration loaded successfully")
+            except Exception as e:
+                self.get_logger().error(f"Error loading config file: {e}")
+                self.los_params = {}
+                self.topic_params = {}
+                self.qos_params = {}
+        else:
+            self.get_logger().warn("No config file path provided. Using defaults.")
+            self.los_params = {}
+            self.topic_params = {}
+            self.qos_params = {}
```

Comment:
This is not needed, and it is not how you should use config files in ROS 2. You correctly declare and get parameters further down.